app/models/church_value.rb

```diff
@@ -1,3 +1,14 @@
 class ChurchValue < ActiveRecord::Base
   belongs_to :church
+
+  def self.values_for(church_ids, period)
+    key = "#{self.class}/#{period}/#{church_ids.join(',')}"
```

Comment:
Rather than church_ids, pass in the array of church objects (assuming you already have it). Then you can use 
`key = "#{self.class}/#{period}/#{churches.collect(&:cache_key).join('/')}"`

The rails generated cache key includes the value of updated_at, which will cause the cache to be invalidated if a church is updated (to Brian's point).